src/components/AccountForm/accountForm.js

```diff
@@ -1,38 +1,71 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import BaseForm from '../BaseForm/baseForm';
 import PropTypes from 'prop-types';
+import { useAuth } from '../../utils/Auth/use-auth';
+import { postAccount } from '../../services/Account/account';
+import { updateAccount } from '../../services/Account/account';
 
-const AccountForm = ({ accountId = null }) => {
-    const [formAccount, setFormAccount] = useState({
-        name: '',
-        amount: 0,
-        type: 1
-    });
+
+const emptyAccount = {
+    uid: '',
+    owner: '',
+    name: '',
+    description: '',
+    balance: 0
+}
+
+const AccountForm = ({ closeForm, selectedAccount }) => {
+    const auth = useAuth();
+    const user = auth?.data;
+    const [formAccount, setFormAccount] = useState(emptyAccount);
+
+    useEffect(() => {
+        updateFormAccount(selectedAccount);
+    }, [selectedAccount]);
 
     const updateFormAccount = (newFormAccount) => {
-        setFormAccount(newFormAccount);
+        if (newFormAccount)
+            setFormAccount(newFormAccount);
     }
 
-    const isEdit = accountId !== null;
+    const isEdit = selectedAccount !== undefined;
 
     const sendRequestToUpdate = () => {
-        alert('update account');
+        updateAccount(selectedAccount.id, {
```

Comment:
It's a good idea to wrap this code into try/catch statement to handle possible errors